lib/help_scout/api/access_token.rb

```diff
@@ -35,10 +59,21 @@ def token_request_params
 
       def initialize(params)
         @value = params[:access_token]
-        @expires_in = params[:expires_in]
-        return unless @expires_in
 
-        @expires_at = Time.now.utc + expires_in
+        if params[:expires_at]
```

Comment:
Curious,  why are we exposing `expires_in` and `expires_at`? Those seem like private implementation details and I don't see them referenced outside of the access token class.

Also, it makes sense that we have a token with an expiry, but having trouble sorting out how the two time values interact. Would you mind clarifying?

Reply:
> Also, it makes sense that we have a token with an expiry, but having trouble sorting out how the two time values interact. Would you mind clarifying?

Sure, and we should probably add some of this to the gem's README or as inline comments.

When you request a token, a successful response will look like this:

```json
{
    "token_type": "bearer",
    "access_token": "369dbb08be58430086d2f8bd832bc1eb",
    "expires_in": 7200
}
```
So this is were `expires_in` comes from. Unfortunately, `expires_in` doesn't do much good after initialization, so we added an `expires_at` to provide a more useful interpretation of the data. 

While working on caching, I noticed that we could instantiate token's with "bad" `expires_at` times if we read the HS token response from a the cache and tried creating an `expirse_at` based on an "old" `expires_in`. To avoid this being an issue, I've updated the `AccessToken` initializer to prefer `expires_at` if it's present (ignoring `expirse_in`).

As for public/private -- before caching, I left these methods public because I thought a host app might like to be able to check the `expires_at` on a token. I agree that `expires_in` can be made private. 

Let me know if there's anything that doesn't make sense here. Happy to discuss it further & would like whatever we add to documentation to be clear & concise.